[Bug] make sync fails locally

### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/kuberay/issues) and found no similar issues.


### KubeRay Component

ray-operator

### What happened + What you expected to happen

**The bug:** When I ran make sync locally I got an error

```
test -s /vagrant/kuberay/ray-operator/bin/kustomize ||  GOBIN=/vagrant/kuberay/ray-operator/bin/kustomize/.. go install sigs.k8s.io/kustomize/kustomize/v3@v3.10.0
go: sigs.k8s.io/kustomize/kustomize/v3@v3.10.0 (in sigs.k8s.io/kustomize/kustomize/v3@v3.10.0):
        The go.mod file for the module providing named packages contains one or
        more exclude directives. It must not contain directives that would cause
        it to be interpreted differently than if it were the main module.
Makefile:117: recipe for target 'kustomize' failed
make: *** [kustomize] Error 1

```
**Expected beahvior**
make sync should work fine and synchronize the required yamls.

**Useful Information**

Go version : go version go1.19.12 linux/amd64
Kustomize install version : v3@v3.10.0

My guess is that it is related to this issue: https://github.com/kubernetes-sigs/kustomize/issues/3618#issuecomment-781521569

I could get around this issue by simply updating the kustomize version to 4.5.2 locally in the Makefile. Supporting comment for this: https://github.com/kubernetes-sigs/kustomize/issues/3618#issuecomment-1054010950

I think the fix is just updating the version of Kustomize. However, I am not confident of what other things are dependent on this (Github CI/CD, other builds, etc). So, opening this issue for investigation. 

### Reproduction script

Navigate to `kuberay/ray-operator`

Run command : `make sync`

or 

`make helm`

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

--- a/ray-operator/Makefile
+++ b/ray-operator/Makefile
@@ -49,7 +49,7 @@ manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and Cust
 generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
 	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
 
-helm: manifests kustomize ## Sync the CRDs into the Helm chart
+helm: manifests ## Sync the CRDs into the Helm chart
 	rm -r ../helm-chart/kuberay-operator/crds/
 	cp -r config/crd/bases/ ../helm-chart/kuberay-operator/crds/
 

